db-client-java/src/main/java/com/eventstore/dbclient/Endpoint.java

```diff
@@ -16,4 +20,13 @@ public String getHostname() {
     public int getPort() {
         return port;
     }
+
+    public URL getURL(boolean secure, String path) {
+        String protocol = secure ? "https" : "http";
+        try {
+            return new URL(protocol + "://" + hostname + ":" + port + path);
```

Comment:
**SUGGESTION:** Minor, but you could use `org.apache.http.client.utils.URIBuilder` to safely build URL.

Reply:
I disagree only because it brings a new dependency for only this line.